### DISCUSSION TITLE
Call from command or batch?

### DISCUSSION THREAD

**Discussion by Oxika95** [Jamminroot/intag#23] [Ideas]
Hello, I just stumbled onto InTag and after a little testing it is absolutely amazing. I have implemented it for myself in my office space and am pushing others to use it as well. Personally I use a bunch of batch scripts to automate job folder setup and have already implemented tags on most of the job folders on our server. I would be really interested in integrating the tag into the batch scripts. is there a way to call intag via command line/batch to assign tags?

**Comment by Jamminroot** [comment]
Hey!
No, for now there is no support for CLI interface (except for install\uninstall), but implementing that sounds like a good idea.
I'll add support for that by the weekend.
For now I'm thinking about something like:
```
-a "tag" # add tag
-r "tag" # remove tag
-f "file" # a file or folder to add or remove tag to\from
-fl "list.txt" # a file with a list of files or folders to add or remove tag to\from
```

For example:
```
intag.exe -a "new_tag" -a "new_tag_2" -r "old_tag" -f "folder1" "fodler2" "folder3"
```

If you have any thoughts, please share

**Comment by Oxika95** [comment]
That sounds great! Again thanks so much! Plan on adding this as one of my
default utilities for windows installs nowadays. Makes me happy because
I've been looking everywhere for this feature set and this is so simple and
just works!

**Comment by Jamminroot** [comment]
I've made a draft of changes, committed it, but not yet tested. Might take some time as I have some stuff to do irl.

**Comment by Oxika95** [comment]
Holy crap! That's so awesome! Can't believe you already got started on
that! Super excited to see these features! Thanks so much, no rush!

**Comment by Jamminroot** [comment]
https://github.com/Jamminroot/intag/releases/tag/1.8.0.0-rc1
I've somewhat fixed problems I faced while testing and made app work with both CLI and UI.
**RE-INSTALLATION IS REQUIRED**

**Reply by Jamminroot** [reply]
Note that bugs might and will likely occur :D Feel free to let me know if you find any

**Comment by Oxika95** [comment]
Dude! Gonna get on this first thing when I get to work tomorrow! Thanks!

**Comment by Oxika95** [comment]
Alright so Ive been messing with this and have gotten it to work. The issue I am currently running into though is the command not being recognized from command even after adding it to path. I have to manually type out the location of the .exe to call it. While that works it means the batch script I would use would be relative to my machine only and fail to tag if anyone else ran the script on their computer. Is there something else other than adding it to path that needs to be done?

**Reply by Jamminroot** [reply]
Current implementation seem to tread "global" calls as if they are calls from explorer for some reason. While forcing it to fall back to "cmd" or "pwsh" might fix CLI, that would break explorer calls.

I'll see if there's anything to do with the problem.

**Reply by Jamminroot** [reply]
Weirdly, I got it to work on my side. What I did is I removed the app from my "tools" folder I have in path, and manually added path to executable folder to PATH back again (separate output folder).
Not exactly sure why, but now app is working. Can you check if setting the app anew works for you?

**Comment by Oxika95** [comment]
Im a little confused as to "added path to executable folder to PATH back again (separate output folder)." I already attempted to manually add a path for in tag, do you mean to remove it and add it back again?
![image](https://github.com/Jamminroot/intag/assets/48342370/85064c84-1dfb-4fc6-8aa0-add36a2fbc40)

**Reply by Oxika95** [reply]
or does it install itself into a location somewhere else? I didint even think to check

**Reply by Oxika95** [reply]
currently using `.\intag.exe --add "test_tag" --path "C:\Users\bryce\Desktop\TestFolder\Folder to Tag"` I get
![image](https://github.com/Jamminroot/intag/assets/48342370/0b4971ee-0bd6-4675-8289-77ffdb91bb9d)
I can get it to work if I specify intags location; "C:\Admin\InTag\intag.exe" --add "test_tag" --path "C:\Users\bryce\Desktop\TestFolder\Folder to Tag"
![image](https://github.com/Jamminroot/intag/assets/48342370/b447f676-c93f-46e3-aca2-d7c9da475657)

**Reply by Jamminroot** [reply]
Yeah, I mean removing old (if any) intag version from PATH, and then re-adding the new version back. I am still fiddling with it, and can't wrap my head around why it actually worked for me, as it doesn't make any sense (or I am completely burnt out and missing smth, lol)

**Reply by Jamminroot** [reply]
As for the ".\\" part yeah, that makes sense - as this prefix would force to use executable from the folder you are currently in. Sorry for confusion

**Reply by Jamminroot** [reply]
... and no, it does not move itself anywhere, only creates entries of itself in registry; moving itself to a different location is not something I'd want (at least my) application to do.

**Reply by Oxika95** [reply]
Okay well I removed the entry closed the path editor and then reopened and added it back again. closed all command windows, relaunched and ran `.\intag.exe --add "tag2" --path "C:\Users\bryce\Desktop\TestFolder\Folder to Tag"` but I got the same error as before. what does your path variable entry look like. should I be adding the "intag.exe" as well or just the containing folder?

**Reply by Jamminroot** [reply]
Just add the containing folder, system will find executable in it.
As for the command, run without ".\\": `intag.exe --add "tag2" --path "C:\Users\bryce\Desktop\TestFolder\Folder to Tag"`
This is what I tested

**Reply by Oxika95** [reply]
that seemed to be the issue, removing `.\` allowed the command to run! I realize now you pointed this out earlier and I didn't register it for some reason. Thank you so much! 
![image](https://github.com/Jamminroot/intag/assets/48342370/0c41d86d-3718-47a1-98ad-e6c96c27ee6a)

**Reply by Oxika95** [reply]
The `.\` mistake seems to expose how much of a novice I am, that seems to have been a misunderstanding on my part from the begging. Thanks for working with me, I appreciate it!

**Reply by Jamminroot** [reply]
No worries, feel free to reach out anytime
